guides/typescript.md

```diff
@@ -0,0 +1,175 @@
+
+Choosing definition styles is almost more of an art than a science. Conciseness and flexibility often go hand in hand, but do not compromise on enforcing correctness to achieve conciseness. Fancy type manipulation with mapped types derived from templated types can be satisfying to figure out, but they often indicate leaky abstractions, vague overloads, and so forth. Use with caution.
+
+Incorrect:
+```ts
+// foo.ts
+export const foo({ a, b, n}: { a: string, b: boolean, n: number}): void { }
+
+// baz.ts
+export interface BazParam {
+  foobar: string
+}
+
+export const baz(bag: BazParam) {}
+
+// bar.ts
+import { foo } from './foo'
+import { BazParam, baz } from './baz'
+
+const bag = {
+  a: 'a',
+  b: false,
+  n: 2
+}
+
+foo(bag)
+
+const bazBag: BazParam = {
+  foobar: 'buzz'
+}
+baz(bazBag)
+```
+
+Better:
+```ts
+// foo.ts
+interface FooParams {
+  a: string
+  b: boolean
+  n: number
+}
+
+export const foo({ a, b, n }: FooParams) {
+}
+
+// bar.ts
+import { foo } from './foo'
+
+const FooParams = Parameters<typeof foo>[0]
+
+const bag: FooParams = { a: 'a', b: false, n: 2 }
+foo(bag)
+```
+
+Best:
+```ts
+// foo.ts
+interface FooParams {
+  a: string
+  b: boolean
+  n: number
+}
+
+export const foo({ a, b, n }: FooParams) {
+}
+
+// bar.ts
+import { foo} from './foo'
+
+foo({
+  a: 'a',
+  b: false,
+  n: 2
+})
+```
+
+#### DRY
+
+If you find yourself defining more than two very similar interfaces or types, consider what can be DRY'd. Is it purely the types, or is there further refactoring that should be done? When DRY, prefer defining the type as close to the implementation that depends on that type as possible. Define parameter bags next to the function they're an argument for, and export them only when necessary. New interfaces should almost never need to be declared for intermediary variables in a function - if you find yourself doing this, there are probably structural refactors that should be considered before DRYing the type definitions.
+
+**Exception to DRY**:
+
+Type declarations composed for public API consumption & definition don't need to be as rigorously DRY'd. Readability and understandability is preferred over conciseness.
+
+## Migration Success Criteria
+
+- No unnecessary `.js` in the repository
+- The repository passes linting based on the `@typescript-eslint/recommended-type-checked` rule set
+- The repository uses `@eslint/stylistic` to format
```

Comment:
`@eslint/stylistic` isn't an npm module

There is a website https://eslint.style/ which refers to plugins `@stylistic/eslint-plugin`

A link either to the website or to an existing npm module would be clearer here